app/controllers/media_wiki_api_controller.rb

```diff
@@ -9,12 +9,10 @@ def api_proxy
               'API proxy (Warning: not for use on public installations!)'
       return render(:json => {error: error})
     end
-    action = api_data.delete(:action)
+    action = params[:action_name]
```

Comment:
Wouldn't we want something like `params.require(:action_name)` (and something similar for `:format`) to avoid this warning from Rails?

```
Unpermitted parameters: :format, :action_name
```

(Arguably dealing with `:format` should be in a separate commit, since this was an extant problem.)

Reply:
This warning shouldn't be causing any issues.

In fact if we did switch to using `params.require(:action_name)` then it could prevent us from returning an error on [line 13](https://github.com/mysociety/verification-pages/pull/43/files#diff-ba4e9df56a14344417b69cc2c0d7b036R13).

The `format` is needed for the client side API but not the API proxy as we always are returning JSON so I think we can safely ignore this one too.